### PR TITLE
Make RSA key check configurable

### LINKED ISSUE #3590: Add a policy action to disable private key checking
We are using push tokens to authenticate  users. The `/triggerchallenge` endpoint calls the `create_challenge()` function in `lib/tokens/pushtoken.py`. This function calls `_build_smartphone_data()` which among other things uses `load_pem_private_key()` to load a per token private key.

Profiling `create_challenge()` for execution time showed that more than 200 msecs are consumed inside `_build_smartphone_data()` with mostly all of them inside `load_pem_private_key()`. Based on this [issue](https://github.com/pyca/cryptography/issues/7236) we tested the following code (a loop of 1000 calls to the function):
```
from cryptography.hazmat.primitives import serialization

for _ in range(1000):
    with open("key.pem", "rb") as key_file:
        private_key = serialization.load_pem_private_key(key_file.read(), password=None)
```
with a total execution time of `29.089s`.

Based on suggestion in the above issue we also added the following on the above code:
```
from cryptography.hazmat.backends.openssl import backend
backend._rsa_skip_check_key = True
```
which resulted in a reduction of total execution time to **0.5 secs**. The same was true if disabling key checking in `_build_smartphone_data()` with a drop of execution time to a few msecs.

Benchmarking the endpoint with and without key checking resulted in a throughtput of 3-4 reqs/sec vs 20-30 reqs/sec depending on concurrent requests. CPU usage was 100% in the first case after 16 concurrent request while it typically remained **below** 100% when key check was disabled. We are talking about an order of magnitude improvement.

Since the private key is created by privacyIDEA itself when a token is enrolled, we would like to suggest that a policy action is added to allow for the key checking to be disabled if desired by the administrator.

Thank you for your time

### PR DESCRIPTION
**Pull request by plettich**
Since v39.0 `pyca/cryptography` support the parameter `unsafe_skip_rsa_key_validation` when loading RSA keys which greatly speeds up the loading time but is unsafe if there is no control over the private key.
The PUSH-token saves the generated private keys encrypted in the database so we can be fairly sure that they were not tampered with.

However for signing the Audit-log as well as signing the response we load the private key from the filesystem.

The added configuration entries `PI_RESPONSE_NO_PRIVATE_KEY_CHECK` and `PI_AUDIT_NO_PRIVATE_KEY_CHECK` (which default to `False`) can be used to disable the costly checking when loading the private keys. These entries should only be enabled if the security of the private keys is assured.

Closes #3590 #3604

**TODO:**
- [x] update documentation

### CODE REVIEW

**Comment by cornelinux** [comment]
@plettich Is this supposed to be draft? Due to TODO-docs?

**Review by cornelinux — COMMENTED** [review]
Looks good to me.
Please also add the docs in 
https://github.com/privacyidea/privacyidea/blob/master/doc/installation/system/inifile.rst

File: doc/faq/crypto-considerations.rst [hunk 1]
```diff
@@ -55,21 +55,25 @@ This way a database administrator is not able to inject rogue password hashes.
...
+
 Audit Signing
 ~~~~~~~~~~~~~
 
-The audit log is digitally signed. (see :ref:`Audit` and :ref:`cfgfile`).
+The audit log is digitally signed. (see :ref:`audit` and :ref:`audit_parameters`).
 
-The audit log can be handled by different modules. privacyIDEA comes with an
-SQL Audit Module.
+The audit log can be handled by different modules. privacyIDEA comes with a
```
Comment (cornelinux) [inline doc/faq/crypto-considerations.rst:65]: @plettich  "an" is correct!

Comment (cornelinux) [reply doc/faq/crypto-considerations.rst:65]: I see. It depends on if SQL is pronounced "esS-Queue-elL" or "sequel".

Comment (plettich) [reply doc/faq/crypto-considerations.rst]: Done in 1ce2fd99b

File: doc/faq/performance.rst [hunk 2]
```diff
@@ -36,7 +36,41 @@ You can configure how often the timestamp should be read using the pi.cfg
...
+safety of the private key in ``PI_AUDIT_KEY_PRIVATE`` can be assured, the config
+entry ``PI_AUDIT_NO_PRIVATE_KEY_CHECK = True`` in :ref:`cfgfile` can improve the performance.
+
+With the config entry ``PI_AUDIT_NO_SIGN = True`` the signing of the Audit-log
+can be deactivated completely.
+
+The privacyIDEA Response
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, privacyIDEA signs every JSON-Response with the private key in ``PI_AUDIT_KEY_PRIVATE``. To improve the performance when loading the private key the config entry ``PI_RESPONSE_NO_PRIVATE_KEY_CHECK`` can be set to ``True``.
```
Comment (cornelinux) [inline doc/faq/performance.rst:71]: @plettich Please put this key in the inifile.rst, this is where we will search for it.

File: doc/faq/crypto-considerations.rst [hunk 3]
```diff
@@ -55,21 +55,25 @@ This way a database administrator is not able to inject rogue password hashes.
...
+SQL Audit Module which is enabled by default.
 
 For signing the audit log the SQL Audit Module uses the RSA keys specified
-with the values *PI_AUDIT_KEY_PUBLIC* and *PI_AUDIT_KEY_PRIVATE* in
+with the values ``PI_AUDIT_KEY_PUBLIC`` and ``PI_AUDIT_KEY_PRIVATE`` in
 :ref:`cfgfile`.
 
 By default the installer generates 2048bit RSA keys.
 
+If the validity of the private key can be assured, the config entry
```
Comment (cornelinux) [inline doc/faq/crypto-considerations.rst:74]: "If the validity of the private key can be assured" and first I did not understand, that you mean, "if you can be sure that the private key is legit or something like this". Maybe you can think of a rephrase. Might not be that necessary.

File: doc/faq/performance.rst [hunk 4]
```diff
@@ -36,7 +36,41 @@ You can configure how often the timestamp should be read using the pi.cfg
...
+
+Please also read :ref:`crypto_considerations` to understand the implications.
+
+.. _faq_perf_crypto_audit:
+
+The Audit-log
+^^^^^^^^^^^^^
+
+Each entry in the :ref:`audit` log is signed to detect tampering. When the
+safety of the private key in ``PI_AUDIT_KEY_PRIVATE`` can be assured, the config
```
Comment (cornelinux) [inline doc/faq/performance.rst:62]: Again. Maybe "if you can be sure, that the private key originates from a trusted source".

**Review by cornelinux — CHANGES_REQUESTED** [review]
My only request is, that you put the two new keys PI_AUDIT_NO_PRIVATE_KEY_CHECK and the other one also in the description for pi.cfg, in inifile.rst. Otherwise we will search endlessly for these keys in a year, when we try to look them up in the docs.

File: doc/faq/performance.rst (hunk 2 above)
Comment (plettich) [reply doc/faq/performance.rst]: Done in 91bcff523

File: doc/faq/crypto-considerations.rst (hunk 3 above)
Comment (plettich) [reply doc/faq/crypto-considerations.rst]: Done in 91bcff523

File: doc/faq/performance.rst (hunk 4 above)
Comment (plettich) [reply doc/faq/performance.rst]: Done in 91bcff523

**Review by cornelinux — APPROVED** [review]

**Pull request merged**